Foundation/include/Poco/ScopedLock.h

```diff
@@ -97,17 +97,25 @@ class ScopedLockWithUnlock
 		}
 	}
 
+	void lock()
+	{
+		poco_assert(_locked == false);
+		_pMutex->lock();
+		_locked = true;
+	}
+
 	void unlock()
 	{
-		if (_pMutex)
+		if (_locked)
```

Comment:
Check for null _pMutex is removed. Would it make sense for both checks to stay? And do the check for nullptr in the ctor also?